runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/SimpleStateRegistry.java

```diff
@@ -56,4 +58,12 @@ public void reset() {
     }
     return monitoringInfos;
   }
+
+  public Map<String, ByteString> getExecutionTimeMonitoringData(ShortIdMap shortIds) {
+    ImmutableMap.Builder<String, ByteString> builder = ImmutableMap.builder();
+    for (SimpleExecutionState state : executionStates) {
+      builder.put(state.getTotalMillisShortId(shortIds), state.getTotalMillisPayload());
```

Comment:
Should we filter out the the state with zero total mills to improve efficiency? It can be common in streaming and I think they don't make a difference.

Reply:
I'll do that as a follow up, after confirming it really doesn't matter.